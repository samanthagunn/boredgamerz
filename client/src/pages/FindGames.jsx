import { useAuth0 } from "@auth0/auth0-react";
import { Loader } from "@googlemaps/js-api-loader";
import { IonContent, IonPage, IonTitle } from "@ionic/react";
import axios from "axios";
import React, { useEffect, useState } from "react";
import AuthHeader from "../components/auth-header";
import Footer from "../components/footer";
import GameList from "../components/game-list";
import FAB from "../components/mobile-fab";
require("dotenv").config();
const FindGames = () => {
  const [data, setData] = useState([]);
  const { getAccessTokenSilently } = useAuth0();
  useEffect(() => {
    getAccessTokenSilently().then((resp) => {
      axios({
        method: "get",
        url: `${process.env.REACT_APP_API_HOST}/game-meeting`,
        headers: {
          Authorization: `Bearer ${resp}`,
        },
      })
        .then((resp) => resp.data)
        .then((data) => {
          setData(data);
        });
    });
    const loader = new Loader({
      apiKey: `${process.env.REACT_APP_MAPS_API_KEY}`,
      version: "weekly",
    });
    loader.load().then(() => {
      new window.google.maps.Map(document.getElementById("map"), {
        center: { lat: 40.0165228, lng: -105.2445022 },
        zoom: 15,
        gestureHandling: "cooperative",
      });
    });
  }, []);
  return (
    <IonPage>
      <AuthHeader />
      <IonContent>
        <br />

        <IonTitle color="light" className="ion-text-center">
          Find Games
        </IonTitle>
        <br />
        <div id="map" style={{ width: "100%", height: "50%" }}></div>
        <GameList seeData={data} joinMode={true} />
      </IonContent>
      <FAB />
      <Footer />
    </IonPage>
  );
};

export default FindGames;
