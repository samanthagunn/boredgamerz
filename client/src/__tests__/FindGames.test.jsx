import React from "react";
import { render, fireEvent, screen } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";
import renderer from "react-test-renderer";
import FindGames from "../pages/FindGames";

describe("The Find Games page...", () => {
    test("will render without crashing", () => {
        const { baseElement } = render(<FindGames />)
        expect(baseElement).toBeDefined();
    });
})