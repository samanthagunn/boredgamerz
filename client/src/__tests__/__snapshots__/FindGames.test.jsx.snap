// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The Find Games page... that is user authenticated... will display correct... 1`] = `
<div
  className="ion-page"
>
  <ion-header>
    <ion-toolbar
      color="primary"
    >
      <a
        href="/home"
      >
        <img
          alt="BoredGamerz Logo"
          className="logo-hero"
          src="https://storage.googleapis.com/boredgamerz_assets/BoredGamerzLogo_72-2.png"
        />
      </a>
      <ion-buttons
        slot="primary"
      >
        <ion-button
          href="/mygames"
          onClick={[Function]}
        >
          My Games
        </ion-button>
        <ion-button
          href="/games"
          onClick={[Function]}
        >
          Find Games
        </ion-button>
        <ion-button
          href="/create"
          onClick={[Function]}
        >
          Create A Game
        </ion-button>
        <ion-button
          onClick={[Function]}
        >
          Profile
        </ion-button>
        <img
          alt="profile"
          className="user-image"
          height="50px"
          onClick={[Function]}
          src="https://lh3.googleusercontent.com/a-/AOh14GiG4KuyPH8-ToSqzoBLwPTvb-px4DboWJn_Gujs=s96-c"
        />
      </ion-buttons>
    </ion-toolbar>
  </ion-header>
  <ion-content>
    <br />
    <ion-title
      color="light"
    >
      Find Games
    </ion-title>
    <br />
    <div
      id="map"
      style={
        Object {
          "height": "50%",
          "width": "100%",
        }
      }
    />
    <ion-item
      onClick={[Function]}
    >
      No Games Found
    </ion-item>
  </ion-content>
  <ion-footer>
    <ion-toolbar
      color="tertiary"
    >
      <div>
        <p>
          Â© 2020 Bored Gamerz, BravoBunch
        </p>
      </div>
    </ion-toolbar>
  </ion-footer>
</div>
`;
